app/form_objects/support/privileged_user_form.rb

```diff
@@ -0,0 +1,45 @@
+class Support::PrivilegedUserForm
+  include ActiveModel::Model
+
+  attr_accessor :full_name, :email_address, :privileges
+
+  validates :full_name, presence: { message: 'Enter the name of the user' }
+  validates :email_address, presence: { message: 'Enter the email address of the user' },
+                            format: { with: /\A.*@computacenter.com|.*@digital.education.gov.uk|.*@education.gov.uk\z/,
+                                      message: 'Email address must be end with @computacenter.com, @digital.education.gov.uk or @education.gov.uk' }
+  validate :validate_email_not_taken
+  validate :validate_atleast_one_privilege
+  validate :validate_acceptable_privileges
+
+  def create_user!
+    User.create!(full_name: full_name, email_address: email_address, is_support: is_support?, is_computacenter: is_computacenter?)
+  end
+
+private
+
+  def is_support?
+    privileges.include?('support')
+  end
+
+  def is_computacenter?
+    privileges.include?('computacenter')
+  end
+
+  def validate_acceptable_privileges
+    if privileges.union(['', 'support', 'computacenter']).size > 3
+      errors.add :privileges, :inclusion, message: 'Only select from available privileges'
+    end
+  end
+
+  def validate_atleast_one_privilege
+    if privileges.reject(&:blank?).empty?
+      errors.add :privileges, :at_least_one, message: 'Select at least one privilege to apply'
+    end
+  end
+
+  def validate_email_not_taken
```

Comment:
Can't this use the standard uniqueness validation?

Reply:
This is not a an ActiveRecord model, so I can't use the standard uniqueness validator as it doesn't hook up to a data source.